testsuite/testcases/src/state_sync_performance.rs

```diff
@@ -52,70 +72,149 @@ impl NetworkTest for StateSyncPerformance {
-        // But, we should probably make it more generic to avoid this.
-        // Ensure we meet the success criteria.
-        let min_expected_tps = ctx.success_criteria.avg_tps as u64;
-        if state_sync_throughput < min_expected_tps {
-            let error_message = format!(
-                "State sync TPS requirement failed. Average TPS: {}, minimum required TPS: {}",
-                state_sync_throughput, min_expected_tps
-            );
-            bail!(error_message)
+        // Generate some traffic through the validators.
+        let emit_txn_duration = ctx.global_duration.checked_div(2).unwrap();
+        info!(
+            "Generating the first round of traffic for {:?} seconds.",
+            emit_txn_duration.as_secs()
+        );
+        let _txn_stat = generate_traffic(ctx, &all_validators, emit_txn_duration, 1)?;
+
+        // Wait for all nodes to synchronize and stabilize.
+        info!("Waiting for the validators to be synchronized.");
+        let runtime = Runtime::new().unwrap();
+        runtime.block_on(async {
+            ctx.swarm()
+                .wait_for_all_nodes_to_catchup(Duration::from_secs(MAX_NODE_LAG_SECS))
+                .await
+        })?;
+
+        // Stop and reset two validators
+        info!("Deleting data for two validators!");
+        let validators_to_restart = &all_validators[0..2];
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.clear_storage().await })?;
+        }
+
+        // Restart the validators so they start syncing from a fresh state
+        for validator_id in validators_to_restart {
+            let validator = ctx.swarm().validator_mut(*validator_id).unwrap();
+            runtime.block_on(async { validator.start().await })?;
         }
 
-        Ok(())
+        // Generate some additional traffic through the validators so the fresh
+        // validators are forced to try and catch up to a moving target.
+        info!(
+            "Generating the second round of traffic for {:?} seconds.",
+            emit_txn_duration.as_secs()
+        );
+        let _txn_stat = generate_traffic(ctx, &all_validators, emit_txn_duration, 1)?;
+
+        // Wait for all nodes to catch up to the highest synced version
+        // then calculate and display the throughput results.
+        calculate_and_display_throughput(ctx, self.name())
     }
 }
+
+/// Calculates and displays the state sync throughput using
+/// the synced version and sync duration.
+fn calculate_and_display_throughput(ctx: &mut NetworkContext<'_>, test_name: &str) -> Result<()> {
+    // Wait for all nodes to catch up to the same synced version
+    let runtime = Runtime::new().unwrap();
+    let timer = Instant::now();
+    let node_sync_duration = ctx.global_duration.saturating_mul(2); // Limits state sync to half consensus throughput
```

Comment:
Do we really need to wait for so long here? global duration is 30 mins, so this is waiting for up to ~ 30 mins.

Reply:
currently, I think forge adds 20 minutes on top of the global_duration, and job gets cleaned up after that (and that is mostly used by starting/stopping things). so if you need to wait a lot, I would divide global_duration by 3 (not 2), and then use two of those thirds for two load tests you have, and one third for waiting here?

@sitalkedia , @rustielin - correct me if I am wrong

(it is defined in forge.py :
`self.global_duration + Duration::from_secs(NAMESPACE_CLEANUP_DURATION_BUFFER_SECS),`

If cluster gets cleaned up , before it finishes, is it guaranteed to show failure? (or can it just silently succeed, without checks finishing?)

Reply:
Yes, good point. The cluster will be cleaned up 20 mins after the global duration set in the test. If it is cleaned up, I think it should show as failed. @rustielin can confirm though.

Reply:
Yea cleanup while the test is expected to run will show up as a failure.

Reply:
This was originally a typo (should have been divided by two). But, I see the overall confusion. I've simplified the test logic for now and I'll return to make this more complex soon. But, both tests now spend 1/2 the test time generating txns, and 1/2 the test time syncing txns. This requires that state sync should be at least as fast as consensus, which makes sense.